### PR TITLE
metrics groupby support

### PR DESCRIPTION
**Pull request by gsps1**
add groupBy support in metrics handler v3 API.
https://issues.cask.co/browse/CDAP-1243

### CODE REVIEW

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java [hunk 1]
```diff
@@ -258,4 +320,21 @@ private void verifySearchResultContains(String url, List<String> expectedValues)
     Assert.assertTrue(reply.containsAll(expectedValues));
   }
 
+  class TimeSeriesResult {
```
Comment (abaranau) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java]: should be separate, anyhow, we'll clenaup with https://github.com/caskdata/cdap/pull/1308

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java [hunk 2]
```diff
@@ -84,6 +85,19 @@ public Cube get() {
       // service
       Constants.Metrics.Tag.SERVICE_RUNNABLE, PROGRAM_LEVEL2
     );
+
+    // reverse mapping
+    // todo : remove hard-coded values and use TypeId (need to move TypeId out of app-fabric ?)
+    this.reverseMapping = ImmutableMap.<String, Map<String, String>>of(
+      //flow
+      "f", ImmutableMap.<String, String>of(PROGRAM_LEVEL2, Constants.Metrics.Tag.FLOWLET,
```
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: we should use TypeId instead of magic strings

**Comment by abaranau** [comment]
one comment - please fix and merge

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java (hunk 2 above)
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: yes , but that adda an dependency on app-fabric, wasn't sure if that's the right thing to do.

**Pull request merged**